File: myth-01-array-insert/python/array_insert.py

from utils.timer_utils.python.timer import Timer

INITIAL_SIZE = 1000000
OPERATION_COUNT = 10000

class ArrayInsert:
  def __init__(self):
    self.list = [];
    self.timer = Timer()

  def prefill_list(self):
    for i in range(INITIAL_SIZE):
      self.list.append(i)
    print(f"Initial size: {len(self.list)}")

  def append_list(self):
    self.timer.start()
    for i in range(OPERATION_COUNT):
      self.list.append(i)
    self.timer.stop("Appending operation took")
    print(f"Final size after appending: {len(self.list)}")

  
  def prepend_list(self):
    self.timer.start()
    for i in range(OPERATION_COUNT):
      self.list.insert(0, i)
    self.timer.stop("Prepending operation took")
    print(f"Final size after prepending: {len(self.list)}")
  



array_insert = ArrayInsert()
array_insert.prefill_list()
array_insert.append_list()
array_insert.prepend_list()

File: myth-01-array-insert/javascript/array-insert.js
import Timer from '../../utils/timer_utils/javascript/timer.js';

const initialSize = 1000000;
const operationCount = 10000;
class ArrayInsert{
  constructor() {
    this.arr = []
  }

  prefillArray() {
    for (let i = 0; i < initialSize; i++) {
      this.arr.push(i);
    }
    console.log(`Initial size: ${this.arr.length}`);
  }

  appendArray() {
    const timer = new Timer();
    timer.start();
    for (let i = 0; i < operationCount; i++) {
      this.arr.push(i);
    }
    timer.stop('Appending operation took');
    console.log(`Final size after appending: ${this.arr.length}`);
  }

  prependArray() {
    const timer = new Timer();
    timer.start();
    for (let i = 0; i < operationCount; i++) {
      this.arr.unshift(i);
    }
    timer.stop('Prepending operation took');
    console.log(`Final size after prepending: ${this.arr.length}`);
  }
}

const ai = new ArrayInsert();
ai.prefillArray();
ai.appendArray();
ai.prependArray();
